File: Helpers/src/main/java/com/dg/helpers/LocationHelper.java
package com.dg.helpers;

/**
 * Created by Daniel Cohen Gindi (danielgindi@gmail.com)
 */
public class LocationHelper
{
    /**
     * Calculates the distance between the specified locations
     * @param lat1
     * @param lon1
     * @param lat2
     * @param lon2
     * @return Distance in kilometers
     */
    public static double getDistanceBetweenCoordinates(
            double lat1, double lon1,
            double lat2, double lon2)
    {
        final double R = 6371.0;
        final double M_PI = Math.PI;
        double DEG_TO_RAD = (M_PI / 180.0);
        lat1 *= DEG_TO_RAD;
        lon1 *= DEG_TO_RAD;
        lat2 *= DEG_TO_RAD;
        lon2 *= DEG_TO_RAD;
        double latDelta = lat2 - lat1;
        double longitudeDelta = lon2 - lon1;

        double a = Math.sin(latDelta/2.0) * Math.sin(latDelta/2.0) +
                Math.cos(lat1) * Math.cos(lat2) *
                        Math.sin(longitudeDelta/ 2.0) * Math.sin(longitudeDelta/2.0);
        double c = 2.0 * Math.atan2(Math.sqrt(a), Math.sqrt(1.0-a));
        return R * c; // Kilometers
    }
}

File: Helpers/src/main/java/com/dg/helpers/LocationHelper.kt
package com.dg.helpers

import kotlin.math.atan2
import kotlin.math.cos
import kotlin.math.sin
import kotlin.math.sqrt

@Suppress("unused")
object LocationHelper
{
    private const val R = 6371.0
    private const val M_PI = Math.PI
    private const val DEG_TO_RAD = M_PI / 180.0

    /**
     * Calculates the distance between the specified locations
     * @param latitude1
     * @param longitude1
     * @param latitude2
     * @param longitude2
     * @return Distance in kilometers
     */
    fun getDistanceBetweenCoordinates(
            latitude1: Double, longitude1: Double,
            latitude2: Double, longitude2: Double): Double
    {
        var lat1 = latitude1
        var lon1 = longitude1
        var lat2 = latitude2
        var lon2 = longitude2

        lat1 *= DEG_TO_RAD
        lon1 *= DEG_TO_RAD
        lat2 *= DEG_TO_RAD
        lon2 *= DEG_TO_RAD
        val latDelta = lat2 - lat1
        val longitudeDelta = lon2 - lon1

        val a = sin(latDelta / 2.0) * sin(latDelta / 2.0) + cos(lat1) * cos(lat2) *
                sin(longitudeDelta / 2.0) * sin(longitudeDelta / 2.0)
        val c = 2.0 * atan2(sqrt(a), sqrt(1.0 - a))
        return R * c // Kilometers
    }
}